crates/turbopack-mdx/src/lib.rs

```diff
@@ -175,14 +179,55 @@ impl MdxTransformedAsset {
             ..Default::default()
         };
 
-        // TODO: upstream mdx currently bubbles error as string
-        let mdx_jsx_component =
-            compile(&file.content().to_str()?, &options).map_err(|e| anyhow!("{}", e))?;
+        let result = compile(&file.content().to_str()?, &options);
 
-        Ok(MdxTransformResult {
-            content: AssetContent::file(File::from(Rope::from(mdx_jsx_component)).into()),
+        match result {
+            Ok(mdx_jsx_component) => Ok(MdxTransformResult {
+                content: AssetContent::file(File::from(Rope::from(mdx_jsx_component)).into()),
+            }
+            .cell()),
+            Err(err) => {
+                let loc = Vc::cell(err.place.map(|p| {
+                    let (start, end) = match *p {
+                        // markdown's positions are 1-indexed, SourcePos is 0-indexed
+                        markdown::message::Place::Position(p) => (
+                            SourcePos {
+                                line: p.start.line - 1,
+                                column: p.start.column - 1,
+                            },
+                            SourcePos {
+                                line: p.end.line - 1,
+                                column: p.end.column - 1,
+                            },
+                        ),
```

Comment:
I still need to figure out whether the `end` of Turbopack's ranges are inclusive or exclusive.

markdown's `end.column` is the first character *after* the range

Reply:
Turbopack's ranges are exclusive as well